flyway-core/src/main/java/org/flywaydb/core/internal/dbsupport/saphana/SapHanaSqlStatementBuilder.java

```diff
@@ -44,22 +40,32 @@ protected String cleanToken(String token) {
 
     @Override
     protected Delimiter changeDelimiterIfNecessary(String line, Delimiter delimiter) {
-        if (StringUtils.countOccurrencesOf(statementStart, " ") < 4) {
```

Comment:
This change actually breaks the case where CREATE and PROCEDURE are on separate lines